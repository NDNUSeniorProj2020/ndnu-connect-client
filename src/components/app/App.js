import React from 'react';

import PreLoginApp from './PreLoginApp';
import PostLoginApp from './PostLoginApp';
import './App.css';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = { loggedIn: !!localStorage.getItem('token'), user: {} };
  }

  handleLogin = (data = {}) => {
    if (data) {
      localStorage.setItem('token', data.token);
      this.setState({ loggedIn: true, user: data.user });
    } else {
      alert('Cannot login.');
    }
  };

  handleLogout = () => {
    localStorage.removeItem('token');
    this.setState({ loggedIn: false, user: {} });
  };

  render() {
    const { handleLogin, handleLogout, state } = this;
    const { loggedIn } = state;

    return <PostLoginApp handleLogout={handleLogout} />;
    //return loggedIn ? <PostLoginApp handleLogout={handleLogout} /> : <PreLoginApp handleLogin={handleLogin} />;
  }
}
