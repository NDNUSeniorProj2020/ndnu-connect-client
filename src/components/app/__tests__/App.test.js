import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

/*
* Don't worry about warning signs in our editor/IDE
* describe comes with Jest which is the default test runner
 */
describe('Testing App component', () => {
	describe('Snapshot tests', () => {
		it('should render without crashing', () => {
			const component = renderer.create(<App />);
			expect(component).toMatchSnapshot();
		});
	});
});