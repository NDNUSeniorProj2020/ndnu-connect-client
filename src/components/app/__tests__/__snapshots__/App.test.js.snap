// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component Snapshot tests should render without crashing 1`] = `ShallowWrapper {}`;
